crates/hir/src/lib.rs

```diff
@@ -4239,6 +4239,20 @@ impl Type {
         }
     }
 
+    pub fn is_fundamental(&self, db: &dyn HirDatabase) -> bool {
+        match self.ty.kind(Interner) {
+            TyKind::Ref(_, _, _) => true,
+            &TyKind::Adt(hir_ty::AdtId(AdtId::StructId(s)), ref _subs) => {
+                db.struct_data(s).flags.contains(StructFlags::IS_FUNDAMENTAL)
+            }
+            _ => false,
+        }
+    }
+
+    pub fn fingerprint_for_trait_impl(&self) -> Option<TyFingerprint> {
+        TyFingerprint::for_trait_impl(&self.ty)
+    }
+
```

Comment:
These aren't needed anymore I think?

Reply:
`is_fundamental` is not needed, but `fingerprint_for_trait_impl` is. `.has_impls_for_trait_and_self_ty()` is defined in `hir-ty`, but `self` (`hir::Type`) is defined in `hir`, and I think you can imagine why I'm not keen on figuring out how to break that circular dependency 😅

Reply:
cool! just removed `is_fundamental` and pushed.